index.js

```diff
@@ -9,13 +9,27 @@ exports.prepare = function (extensions, filepath, cwd, nothrow) {
   var attempts = [];
   var err;
   var onlyErrors = false;
-  var ext = extension(filepath);
-  if (Object.keys(require.extensions).indexOf(ext) !== -1) {
+  var exts = extension(filepath);
+
+  var supportsExtension = Object.keys(require.extensions).indexOf(exts[0]) !== -1;
```

Comment:
Well, if you look at the validation I'm using only the first option `exts[0]` (e.g `.babel.js`). If we don't do this check we would treat `.babel.js` as `.js` files. You can check that by removing it and running the tests.